### ISSUE TITLE
Complete obsoletion process for DomainUpDownAccessibleObject class.

### ISSUE THREAD

**Issue by Tanya-Solyanik** [dotnet/winforms#7587]
We'll be conservative in NET7 and will return instance of an obsoleted DomainUpDownAccessibleObject type from DomainUpDown.CreateAccessibilityInstance method. This will prevent failures of the runtime casts in snippets like this one:
```
internal class MySpinner : DomainUpDown
{
<snip>
DomainUpDownAccessibleObject ao = CreateAccessibleObject() as DomainUpDownAccessibleObject;
<snip>
}
```

In .NET 8 we will remove protected override methos `DomainUpDown.CreateAccessibilityInstance`. This kind of change is allowed by the [breaking changes](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md#members) rules:
```
Introducing or removing an override

Make note, that introducing an override might cause previous consumers to skip over the override when calling base.
```

Additionally, we should consider promoting build time warning to an error because this class had always been documented as internal use only. Lets discuss this part of the change within the team.

_Originally posted by @Tanya-Solyanik in https://github.com/dotnet/winforms/pull/7330#discussion_r944772133_

**Comment by dmitrii-drobotov** [comment]
Created issue https://github.com/dotnet/docs/issues/32507 to update the documentation

**Comment by MelonWang1** [comment]
Verified on latest 8.0 build .NET 8.0.100-alpha.1.22607.6, issue is fixed. Screenshot as below:
![Screenshot 2022-12-07 190256](https://user-images.githubusercontent.com/94418985/206346670-c89f21e9-8ef8-44b8-8ff1-aaf2d9bca961.png)
